lib/optimize/SideEffectsFlagPlugin.js

```diff
@@ -9,12 +9,11 @@ const HarmonyExportImportedSpecifierDependency = require("../dependencies/Harmon
 const HarmonyImportSideEffectDependency = require("../dependencies/HarmonyImportSideEffectDependency");
 const HarmonyImportSpecifierDependency = require("../dependencies/HarmonyImportSpecifierDependency");
 
-const evaluateWithFalseValue = (sideEffects, file) => {
-	return (sideEffects === "false") ?
-		// Assume we mean (Boolean) false, except for files/modules named false.
-		(/false.*$/).test(file) :
-		sideEffects !== false;
-};
+const evaluateWithFalseValue = (sideEffects, file) =>
+	(sideEffects === "false") ?
```

Comment:
This is ugly and doesn't quite match the other code, but it was the only thing I could find to make both the linters happy.